daemon/cmd/watchdogs.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/datapath/loader"
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+const epBPFProgWatchdog = "ep-bpf-prog-watchdog"
+
+type epBPFProgWatchdogConfig struct {
+	EndpointBPFProgWatchdogInterval time.Duration
+	EndpointBPFProgWatchdogEnabled  bool
+}
+
+func (c epBPFProgWatchdogConfig) Flags(flags *pflag.FlagSet) {
+	flags.Bool("endpoint-bpf-prog-watchdog-enabled", c.EndpointBPFProgWatchdogEnabled,
+		"Enable endpoint BPF program watchdog")
+
+	flags.Duration("endpoint-bpf-prog-watchdog-rate", c.EndpointBPFProgWatchdogInterval,
+		"Interval to trigger endpoint BPF programs load check watchdog")
+}
+
+type epBPFProgWatchdogParams struct {
+	cell.In
+
+	Config        epBPFProgWatchdogConfig
+	Logger        logrus.FieldLogger
+	Lifecycle     hive.Lifecycle
+	DaemonPromise promise.Promise[*Daemon]
+}
+
+var (
+	// EndpointBPFrogWatchdogCell triggers a job to ensure device tc programs remain loaded.
+	EndpointBPFrogWatchdogCell = cell.Module(
```

Comment:
I would reference this from `daemonCell` to keep modules building on top of `promise.Promise[*Daemon]` under it.

Reply:
Indeed!